transport/default_mapper.go

```diff
@@ -0,0 +1,89 @@
+package transport
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/ThreeDotsLabs/esja/stream"
+)
+
+// Event is a transport model which defines which stream model
+// it corresponds to and implements the mapping from- and to- the stream model.
+type Event[T any] interface {
+	SupportedEvent() stream.Event[T]
+	FromEvent(event stream.Event[T]) Event[T]
+	ToEvent() stream.Event[T]
```

Comment:
I think we might need to work on these names, as now it converts an event to an event. 

Perhaps something like `FromTransport` / `ToTransport` like in `Mapper`?

Reply:
Since the interface implements the transport model it looks pretty odd when you implement the method `ToTransport(streamEvent)` of the transport event. The name suggests that this method should return something mapped into a transport model. Similarily, implementing `FromTransport() Event` you expect some input, don't you? Maybe to make it event clearer it could be `FromStreamEvent / ToStreamEvent`?

Reply:
https://github.com/ThreeDotsLabs/esja/pull/18/commits/e25796691ea4a971492d565d9b6f71d6cabb0801

Reply:
nice!

Reply:
Makes sense 👍